pkg/reconciler/v1alpha1/taskrun/taskrun_test.go

```diff
@@ -197,18 +211,136 @@ func TestReconcileBuildCreationErrors(t *testing.T) {
+	}
+
+	c, _, clients := getController(d)
+
+	reactor := func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+		if action.GetVerb() == "get" && action.GetResource().Resource == "builds" {
+			// handled fetching builds
+			return true, nil, fmt.Errorf("induce failure fetching builds")
+		}
+		return false, nil, nil
+	}
+
+	clients.bclient.PrependReactor("*", "*", reactor)
+
+	if err := c.Reconciler.Reconcile(context.Background(), fmt.Sprintf("%s/%s", taskRun.Namespace, taskRun.Name)); err == nil {
+		t.Fatal("expected error")
+	}
+}
+
+func TestReconcileBuildUpdateStatus(t *testing.T) {
+	taskRun := &v1alpha1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-taskrun-run-success",
+			Namespace: "foo",
+		},
+		Spec: v1alpha1.TaskRunSpec{
+			TaskRef: v1alpha1.TaskRef{
+				Name:       "test-task",
+				APIVersion: "a1",
+			},
+		},
+	}
+	d := testData{
+		taskruns: []*v1alpha1.TaskRun{
+			taskRun,
+		},
+		tasks: []*v1alpha1.Task{simpleTask},
+	}
+	buildSt := &duckv1alpha1.Condition{
+		Type: duckv1alpha1.ConditionSucceeded,
+		// build is not completed
+		Status:  corev1.ConditionUnknown,
+		Message: "Running build",
+	}
+
+	c, _, clients := getController(d)
+	build := &buildv1alpha1.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      taskRun.Name,
+			Namespace: taskRun.Namespace,
+		},
+		Spec: *simpleTask.Spec.BuildSpec,
+	}
+	build.Status.SetCondition(buildSt)
+
+	_, err := clients.bclient.BuildV1alpha1().Builds(taskRun.Namespace).Create(build)
+	if err != nil {
+		t.Errorf("error creating build : %v", err)
+	}
+
+	if err := c.Reconciler.Reconcile(context.Background(), fmt.Sprintf("%s/%s", taskRun.Namespace, taskRun.Name)); err != nil {
+		t.Fatal("Unexpected error", err)
+	}
+
+	newTr, err := clients.taskrunInformer.Lister().TaskRuns(taskRun.Namespace).Get(taskRun.Name)
+	if err != nil {
+		t.Errorf("error: %v", err)
+	}
+	var ignoreLastTransitionTime = cmpopts.IgnoreTypes(duckv1alpha1.Condition{}.LastTransitionTime.Inner.Time)
+	if d := cmp.Diff(newTr.Status.GetCondition(duckv1alpha1.ConditionSucceeded), buildSt, ignoreLastTransitionTime); d != "" {
+		t.Fatalf("-want, +got: %v", d)
+	}
+
+	// update build status and trigger reconcile
+	buildSt.Status = corev1.ConditionTrue
+	buildSt.Message = "Build completed"
+
+	build.Status.SetCondition(buildSt)
+
+	_, err = clients.bclient.BuildV1alpha1().Builds(taskRun.Namespace).Update(build)
+	if err != nil {
+		t.Errorf("Unexpected error creating build: %v", err)
+	}
+
+	if err := c.Reconciler.Reconcile(context.Background(), fmt.Sprintf("%s/%s", taskRun.Namespace, taskRun.Name)); err != nil {
+		t.Fatalf("Unexpected error when Reconcile(): %v", err)
+	}
+
+	newTr, err = clients.taskrunInformer.Lister().TaskRuns(taskRun.Namespace).Get(taskRun.Name)
+	if err != nil {
+		t.Errorf("Unexpected error fetching taskrun: %v", err)
+	}
+	if d := cmp.Diff(newTr.Status.GetCondition(duckv1alpha1.ConditionSucceeded), buildSt, ignoreLastTransitionTime); d != "" {
```

Comment:
this should not be executed if error is not nil. Maybe use t.Fatalf before or do this in else.